pinot-controller/src/test/java/org/apache/pinot/controller/ControllerConfTest.java

```diff
@@ -0,0 +1,158 @@
+          REALTIME_SEGMENT_VALIDATION_FREQUENCY_PERIOD, REALTIME_SEGMENT_VALIDATION_INITIAL_DELAY_PERIOD,
+          BROKER_RESOURCE_VALIDATION_FREQUENCY_PERIOD, BROKER_RESOURCE_VALIDATION_INITIAL_DELAY_PERIOD,
+          STATUS_CHECKER_FREQUENCY_PERIOD, STATUS_CHECKER_WAIT_FOR_PUSH_TIME_PERIOD, TASK_MANAGER_FREQUENCY_PERIOD,
+          MINION_INSTANCES_CLEANUP_TASK_FREQUENCY_PERIOD, MINION_INSTANCES_CLEANUP_TASK_INITIAL_DELAY_PERIOD,
+          MINION_INSTANCES_CLEANUP_TASK_MIN_OFFLINE_TIME_BEFORE_DELETION_PERIOD, TASK_METRICS_EMITTER_FREQUENCY_PERIOD,
+          SEGMENT_RELOCATOR_FREQUENCY_PERIOD, SEGMENT_LEVEL_VALIDATION_INTERVAL_PERIOD,
+          STATUS_CHECKER_INITIAL_DELAY_PERIOD, OFFLINE_SEGMENT_INTERVAL_CHECKER_INITIAL_DELAY_PERIOD,
+          SEGMENT_RELOCATOR_INITIAL_DELAY_PERIOD);
+
+  private static final int DURATION_IN_SECONDS = 20;
+  private static final String DURATION_IN_PERIOD_VALID = "2m";
+  private static final int PERIOD_DURATION_IN_SECONDS = 120;
+  private static final String DURATION_IN_PERIOD_INVALID = "2m2m";
+
+  /**
+   * When both type of configs are present, then the human-readable period config overrides the unit config
+   */
+  @Test
+  public void periodConfigOverridesUnitConfig() {
+    //setup
+    Map<String, Object> controllerConfig = new HashMap<>();
+    UNIT_CONFIGS.forEach(config -> controllerConfig.put(config, DURATION_IN_SECONDS));
+    PERIOD_CONFIGS.forEach(config -> controllerConfig.put(config, DURATION_IN_PERIOD_VALID));
+    ControllerConf conf = new ControllerConf(controllerConfig);
+    //execution and assertion
+    assertOnDurations(conf, PERIOD_DURATION_IN_SECONDS);
+  }
+
+  /**
+   * When only unit configs are supplied, then the correct converted value is returned.
+   */
+  @Test
+  public void suppliedUnitConfigsShouldReturnCorrectValues() {
+    //setup
+    Map<String, Object> controllerConfig = new HashMap<>();
+    UNIT_CONFIGS.forEach(config -> controllerConfig.put(config, DURATION_IN_SECONDS));
+    ControllerConf conf = new ControllerConf(controllerConfig);
+    //execution and assertion
+    assertOnDurations(conf, DURATION_IN_SECONDS);
+  }
+
+  /**
+   * When only period configs are supplied, then the correct converted value is returned.
+   */
+  @Test
+  public void suppliedPeriodConfigsShouldReturnCorrectValues() {
+    //setup
+    Map<String, Object> controllerConfig = new HashMap<>();
+    PERIOD_CONFIGS.forEach(config -> controllerConfig.put(config, DURATION_IN_PERIOD_VALID));
+    ControllerConf conf = new ControllerConf(controllerConfig);
+    //execution and assertion
+    assertOnDurations(conf, PERIOD_DURATION_IN_SECONDS);
+  }
+
+  /**
+   * When valid unit config and invalid period config are specified, then an {@link IllegalArgumentException} is thrown (valid unit
+   * config does not override invalid period config)
+   */
+  @Test(expectedExceptions = IllegalArgumentException.class)
```

Comment:
Expected exceptions are good, but I prefer fielding specific exceptions at the specific points where we expect them. That makes it easy to add more code to the test, or re-factor things, etc. and not worry about the test passing if we accidentally get IllegalArgumentException some place else.

Reply:
That's a valid point @mcvsubbu. I just went with the simplest approach as the test was very trivial, and you pretty much can point where exactly the exception was thrown. Will change it.